src/toolkit/matl_buy_policy.cc

```diff
@@ -186,11 +253,17 @@ void MatlBuyPolicy::Stop() {
 std::set<RequestPortfolio<Material>::Ptr> MatlBuyPolicy::GetMatlRequests() {
   rsrc_commods_.clear();
   std::set<RequestPortfolio<Material>::Ptr> ports;
-  bool make_req = buf_->quantity() < req_when_under_ * buf_->capacity();
+
+  // (s,S)/(R,Q) inventory policy handling
+  if (!MakeReq()) {
+    return ports;
+  }
```

Comment:
Is there a specific code logic reason this moved up here?

Reply:
Yeah, you don't need to calculate anything below if MakeReq is false, so it saves executing useless code if that's the case